Dockerfile

```diff
@@ -1,16 +1,26 @@
-FROM golang:latest AS builder
+# Vipnode binary-builder image using Alpine.
+#
+# To run a vipnode agent, you'll need to volume-mount the IPC socket or expose
+# the appropriate RPC ports to connect to your node.
+#
+# Example:
+#   docker build . --tag "vipnode"
+#   docker run --rm -it vipnode --help
 
+# Builder environment
+FROM golang:alpine AS builder
+RUN apk update && apk --no-cache add git gcc build-base
+
+# Build the source
 COPY . /go/src/vipnode
 WORKDIR /go/src/vipnode
 RUN make
 
-# Copy binary to a fresh container, to avoid including build artifact.
-
-# FIXME: Would be nice to use alpine (reduces the image size from >700mb to
-# <30mb, but we'd need to statically link which is tricky with the Keccak256 C
-# lib sub-dependency (via geth).
-FROM golang:latest
-COPY --from=builder /go/src/vipnode/vipnode /vipnode
+# Run environment
+FROM alpine
+RUN apk update && apk --no-cache add ca-certificates
```

Comment:
very minor: TIL that `ADD` can take a URL; I've seen some Dockerfiles use this instead to get the CA certs:

```
ADD https://curl.haxx.se/ca/cacert.pem /etc/ssl/certs/ca-certificates.crt
```

See for example: https://hub.docker.com/r/networld/grpcurl/dockerfile

Anyways only mention as it's the only reason apk is being used, so could make that layer even smaller if we wanted to "Docker golf" the image. :)

Reply:
Mmmm that is cute! will keep it in mind. :D